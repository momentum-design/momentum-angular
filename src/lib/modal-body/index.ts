export * from './modal-body.module';
export * from './modal-body.component';
