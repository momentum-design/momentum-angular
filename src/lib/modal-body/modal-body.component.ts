import { Component, OnInit, Input, HostBinding } from '@angular/core';

@Component({
  selector: 'cui-modal-body',
  template: `
    <ng-content></ng-content>
  `,
  styles: []
})
export class ModalBodyComponent implements OnInit {

  /** @option css class names | '' | '' */
  @Input() public class: string = '';

  @HostBinding('class') get className(): string {
    return 'cui-modal__body' +
    `${(this.class && ` ${this.class}`) || ''}` +
    ``;
  }

  constructor() { }

  ngOnInit() {}

}

/**
 * @component modal-body
 * @section class
 * @angular
 *
<cui-modal>
  <cui-modal-body class='large'></cui-modal-body>
</cui-modal>
 */

/**
 * @component modal-body
 * @section ng-content
 * @angular
 *
<cui-modal>
  <cui-modal-body class='large'>
    <p>MyBody Dom...</p>
  </cui-modal-body>
</cui-modal>
 */
