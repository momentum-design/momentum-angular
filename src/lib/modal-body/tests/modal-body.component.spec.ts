import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ModalBodyComponent } from '../modal-body.component';

describe('ModalBodyComponent', () => {
  let component: ModalBodyComponent;
  let fixture: ComponentFixture<ModalBodyComponent>;
  let testElement: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ModalBodyComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ModalBodyComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render classname', () => {
    component.class = 'textAddOn';
    fixture.detectChanges();
    testElement = fixture.nativeElement;
    expect(testElement.className).toContain('textAddOn');
  });
});
