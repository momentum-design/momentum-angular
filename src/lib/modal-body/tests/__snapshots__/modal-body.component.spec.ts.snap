// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBodyComponent should create and match snapshot 1`] = `
<md-modal-body
  class=""
>

</md-modal-body>
`;
